cli/src/git_util.rs

```diff
@@ -172,17 +176,140 @@ pub fn with_remote_git_callbacks<T>(
     f(callbacks)
 }
 
-pub fn print_git_import_stats(ui: &mut Ui, stats: &GitImportStats) -> Result<(), CommandError> {
+pub fn print_git_import_stats(
+    ui: &mut Ui,
+    repo: &dyn Repo,
+    stats: &GitImportStats,
+    show_ref_stats: bool,
+) -> Result<(), CommandError> {
+    if show_ref_stats {
+        let refs_stats = stats
+            .changed_remote_refs
+            .iter()
+            .map(|(ref_name, (remote_ref, ref_target))| {
+                RefStatus::new(ref_name, remote_ref, ref_target, repo)
+            })
+            .collect_vec();
+
+        let has_both_ref_kinds = refs_stats
+            .iter()
+            .any(|x| matches!(x.ref_kind, RefKind::Branch))
+            && refs_stats
+                .iter()
+                .any(|x| matches!(x.ref_kind, RefKind::Tag));
+
+        let max_width = refs_stats.iter().map(|x| x.ref_name.width()).max();
+        if let Some(max_width) = max_width {
+            let mut stderr = ui.stderr_formatter();
+            for status in refs_stats {
+                status.output(max_width, has_both_ref_kinds, &mut *stderr)?;
+            }
+        }
+    }
+
     if !stats.abandoned_commits.is_empty() {
         writeln!(
             ui.stderr(),
             "Abandoned {} commits that are no longer reachable.",
             stats.abandoned_commits.len()
         )?;
     }
+
     Ok(())
 }
 
+struct RefStatus {
+    ref_kind: RefKind,
+    ref_name: String,
+    tracking_status: TrackingStatus,
+    import_status: ImportStatus,
+}
+
+impl RefStatus {
+    fn new(
+        ref_name: &RefName,
+        remote_ref: &RemoteRef,
+        ref_target: &RefTarget,
+        repo: &dyn Repo,
+    ) -> Self {
+        let mut tracking_status = TrackingStatus::Untracked;
+        let (ref_name, ref_kind) = match ref_name {
+            RefName::RemoteBranch { branch, remote } => {
+                if repo.view().get_remote_branch(branch, remote).is_tracking() {
+                    tracking_status = TrackingStatus::Tracked;
+                }
+                (format!("{branch}@{remote}"), RefKind::Branch)
+            }
+            RefName::Tag(tag) => {
+                tracking_status = TrackingStatus::NotApplicable;
+                (tag.clone(), RefKind::Tag)
+            }
+            RefName::LocalBranch(branch) => (branch.clone(), RefKind::Branch),
```

Comment:
nit: set/return `(ref_name, ref_kind, tracking_status)` by `LocalBranch` consistently?

Under the hood, `Tag` and `LocalBranch` (which is `@git` branch) are always `Tracked`. We can of course map these to `NotApplicable` to suppress redundant output.